gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagStateStoreWithDagNodes.java

```diff
@@ -40,7 +40,8 @@ public interface DagStateStoreWithDagNodes extends DagStateStore {
    * Returns 1 if the dag node is inserted as a new one, 2 if is updated, and 0 if new dag node is same as the existing one
    * <a href="https://dev.mysql.com/doc/refman/8.4/en/insert-on-duplicate.html">Refer</a>
    */
-  int updateDagNode(DagManager.DagId dagId, Dag.DagNode<JobExecutionPlan> dagNode) throws IOException;
+  int updateDagNode(DagManager.DagId dagId, Dag.DagNode<JobExecutionPlan> dagNode, boolean isFailedDag) throws IOException;
```

Comment:
isFailedDag is also inside DagNode, right? do we still need isFailedDag as a separate parameter?

Reply:
We have removed from DagNode, so it's required now

Reply:
wondering on semantics then.  here we have a method to update a `DagNode`, but it takes a flag about whether its parent, the DAG is failed.  does `true` here mean, in the course of updating the `DagNode`, to mark the entire DAG failed?  (I can imagine that faillure in one of the nodes would render the entire DAG failed.)

that said, what if there are multiple concurrent jobs, each corresponding to a `DagNode`, and they all complete about the same time, with several succeeding, but one failing?  is there a race-condition where unless the failed `DagNode` is processed last, that a successful `DagNode` might get processed just after the failure and re-set the DAG back to NOT failed?